File: app/assets/javascripts/Series.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const SeriesSchema = DataElementSchema({
  uid: Any,
  number: Any,
  modality: Any,
  description: Any,
  numberOfInstances: Any,
  endpoint: Any,
  bodySite: Any,
  laterality: Any,
  specimen: Any,
  started: Any,
  performer: Any,
  instance: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'Series' },

});

module.exports.SeriesSchema = SeriesSchema;
module.exports.Series = mongoose.model('Series', SeriesSchema);

File: app/models/qdm/series.rb
module QDM
  # app/models/qdm/series.rb
  class Series < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :uid
    field :number
    field :modality
    field :description
    field :numberOfInstances
    field :endpoint
    field :bodySite
    field :laterality
    field :specimen
    field :started
    field :performer
    field :instance
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
